File: basic/1008.cpp
#include <iostream>
using namespace std;

const int maxn = 105;
int a[maxn];

int main() {
    int n, m;
    cin >> n >> m;
    m %= n;
    for (int i = 0; i < n; ++i) {
        cin >> a[i];
    }
    for (int i = 0; i < n; ++i) {
        if (i)
            cout << " ";
        cout << a[(i - m + n) % n];
    }
    cout << endl;
    return 0;
}

File: basic/1008.java
import java.util.*;

class Main {
    public static final int maxn = 105;
    public static int[] a = new int[maxn];

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int m = in.nextInt() % n;
        for (int i = 0; i < n; ++i) {
            a[i] = in.nextInt();
        }
        for (int i = 0; i < n; ++i) {
            if (i > 0) {
                System.out.print(" ");
            }
            System.out.print(a[(i + n - m) % n]);
        }
        System.out.print("\n");

        in.close();
    }
}